[Security Solution] Bugfix: Hide rule link, alerts table on case-attached alerts

rylnd opened the pull request:
## Summary

This addresses some bugs between Cases and Rules RBAC: We are in essence trying to pass Rules RBAC information to Cases components, but this is not currently well-supported as these live in separate plugins, and the client  context/hooks that we've been developing for Rules/Alerts RBAC are not consumable by these Cases components (particularly, I'd like to use `useAlertsPrivileges` within `x-pack/platform/plugins/shared/cases/public/components/user_actions/comment/alert_event.tsx`). For now, this is the best we can do without passing that information across plugins.

### Details

* Addresses https://github.com/elastic/kibana/issues/247049 by hiding the alerts table when the users does not have read access to alerts (and instead showing a privileges callout):
    <kbd>
 
<img width="851" height="865" alt="Screenshot 2026-01-20 at 10 03 20 PM" src="https://github.com/user-attachments/assets/68f864e0-82b7-4cfe-b60c-bc8fc4e93f63" />

    </kbd>



* (partially) addresses https://github.com/elastic/kibana/issues/247046 by not opening the flyout when users click on a rule name from within a case:

    https://github.com/user-attachments/assets/bffc9deb-5322-4774-95f1-be6405dd1fff


    * Note that a 403 still occurs when rule details are attempted to be fetched on the Alert Details flyout; I did not have time yet to address this case.

### Checklist

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [x] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.






<!--ONMERGE {"backportTargets":["9.3"]} ONMERGE-->

rylnd committed:
Hide Alerts table on Cases page if user cannot read Alerts

This hides both the EASE alerts table, and the regular alerts table, if
the user does not have sufficient RBAC privileges to retrieve the alerts
in question.

Addresses #247049.

rylnd committed:
Do not open flyout for Cases Rule if user cannot read Rules

I cannot currently change the styling of the rule name, as that is
generated within the cases plugin and has no knowledge of rule
privileges/access.

Partially addresses #247046.

rylnd committed:
Merge branch 'main' into hide_rule_link_on_attached_alerts

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/377616)
* Commit: e64332a5bb52252bc2fb0766a17de67bba91a76b

### Failed CI Steps
* [FTR Configs #38](https://buildkite.com/elastic/kibana-pull-request/builds/377616#019b8087-173f-4c20-ba28-49d33d625004)
* [Scout: [ observability / apm ] plugin](https://buildkite.com/elastic/kibana-pull-request/builds/377616#019b808c-e113-4bdc-b38d-5df063f1855d)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/377616#019b808c-e113-4bdc-b38d-5df063f1855d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/377616/jobs/019b808c-e113-4bdc-b38d-5df063f1855d/artifacts/019b809d-678d-4436-aa2d-39b169b6f227) Scout: [ observability / apm ] plugin / Can create, trigger and view an 'Error count' alert from service inventory
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/377616#019b8087-173f-4c20-ba28-49d33d625004) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/377616/jobs/019b8087-173f-4c20-ba28-49d33d625004/artifacts/019b80ae-c37d-4873-9137-3abc066dbb51) FTR Configs #<span></span>38 / Cloud Security Posture Security Network Page - Graph visualization ECS fields only expanded flyout - entity enrichment for multiple actors and targets
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/377616#019b808c-e113-4bdc-b38d-5df063f1855d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/377616/jobs/019b808c-e113-4bdc-b38d-5df063f1855d/artifacts/019b809d-678d-44a6-95eb-23fa7ef06657) Scout: [ observability / apm ] plugin / serverless-oblt - Alerts - Can create, trigger and view an 'Error count' alert from service inventory

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/6d9c98f936253cd6c0fa7d8444b5620dea02c7c8) | [after](https://github.com/elastic/kibana/commit/e64332a5bb52252bc2fb0766a17de67bba91a76b) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 10.7MB | 10.7MB | +81.0B |



cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377616","number":377616,"commit":"e64332a5bb52252bc2fb0766a17de67bba91a76b"}],"number":377616}
buildkite-pr-comment-->

rylnd committed:
Merge branch 'main' into hide_rule_link_on_attached_alerts

rylnd committed:
Merge branch 'main' into hide_rule_link_on_attached_alerts

rylnd committed:
Fix various interactions with alerts permissions in cases

* Hides "sync alerts" button when appropriate
* Disables the rule link when appropriate
* Prevents users from deleting alerts when appropriate

rylnd committed:
Display "Insufficient Privileges" callout in lieu of Alerts table

When viewing a case with attached alerts, but lacking alert privileges,
a callout is now displayed instead of the alerts table.

rylnd committed:
Merge branch 'main' into hide_rule_link_on_attached_alerts

rylnd committed:
Set default values for alerts RBAC in cases context

This will mimic the previous behavior for consumers: the alerts paths
will be active and data will be reliant on the user's ES permissions.

In the future, consumers (o11y, stack) can override these values with their
appropriate RBAC permissions.

rylnd committed:
Add missing keys to mock context values in tests

This resolves a few type errors and should produce a green build.

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/384563)
* Commit: e1820a0f1965257452156ca31f74f5d145e2860c

### Failed CI Steps
* [FTR Configs #129](https://buildkite.com/elastic/kibana-pull-request/builds/384563#019be203-f20c-4c17-8cfb-8e322f41f25d)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/871ba7b27e769f3c760d5a636ce1ae2133ae603d) | [after](https://github.com/elastic/kibana/commit/e1820a0f1965257452156ca31f74f5d145e2860c) | diff |
| --- | --- | --- | --- |
| `cases` | 1.4MB | 1.4MB | +68.0B |
| `securitySolution` | 10.8MB | 10.8MB | +215.0B |
| total |  |  | +283.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/871ba7b27e769f3c760d5a636ce1ae2133ae603d) | [after](https://github.com/elastic/kibana/commit/e1820a0f1965257452156ca31f74f5d145e2860c) | diff |
| --- | --- | --- | --- |
| `cases` | 144.3KB | 144.3KB | +15.0B |



### History
* :broken_heart: [Build #384086](https://buildkite.com/elastic/kibana-pull-request/builds/384086) failed 2cd5b73913c447f6aefa0a0996cebb1beea7e40c
* :yellow_heart: [Build #377616](https://buildkite.com/elastic/kibana-pull-request/builds/377616) was flaky e64332a5bb52252bc2fb0766a17de67bba91a76b

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/384563","number":384563,"commit":"e1820a0f1965257452156ca31f74f5d145e2860c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/384086","number":384086,"commit":"2cd5b73913c447f6aefa0a0996cebb1beea7e40c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377616","number":377616,"commit":"e64332a5bb52252bc2fb0766a17de67bba91a76b"}],"number":384563}
buildkite-pr-comment-->

michaelolo24 reviewed (approved):
Thanks for making the changes!

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|9.3|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/250002">](https://github.com/elastic/kibana/pull/250002)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->
